tools/clang/include/clang/Sema/SemaHLSL.h

```diff
@@ -128,6 +128,8 @@ unsigned CaculateInitListArraySizeForHLSL(clang::Sema *sema,
                                           const clang::InitListExpr *InitList,
                                           const clang::QualType EltTy);
 
+bool HasLongVecs(const clang::QualType &qt);
```

Comment:
nit: IIUC, this is really checking if the type passed in contains a sub-object of long vector type?
```suggestion
bool ContainsLongVecSubObject(const clang::QualType &);
```

Reply:
I didn't spend too much time considering this name. With other changes, I see this taking a max length param which makes the "long" in "longvec" subjective, but I'll name it something similar to your suggestion.

Reply:
I don't think it should take a maximum length parameter, isn't a "long" vector anything over 4?